### PR TITLE
Add thirdparty.json

### PR DESCRIPTION
**Pull request by baothientran**
This is similar to Cesium for Unreal's ThirdParty json

### CODE REVIEW

File: ThirdParty.json [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    "url": "https://github.com/microsoft/GSL",
+    "license": "MIT"
+  },
+  {
+    "name": "KTX-Software",
+    "url": "https://github.com/KhronosGroup/KTX-Software",
+    "license": "Apache 2.0"
+  },
+  {
+    "name": "MikkTSpace",
```
Comment (shehzan10) [inline ThirdParty.json:63]: @baothientran Do we depend on MilkkTSpace in O3DE?

File: ThirdParty.json [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    "name": "SQLite Amalgamation",
+    "url": "https://www.sqlite.org/amalgamation.html",
+    "license": "Public Domain"
+  },
+  {
+    "name": "STB",
+    "url": "https://github.com/nothings/stb",
+    "license": "Public Domain"
+  },
+  {
```
Comment (shehzan10) [inline ThirdParty.json:102]: @baothientran We should also include tidyhtml2 right? https://github.com/CesiumGS/cesium-o3de/tree/main/External. Are there any others we may have missed?

**Review by shehzan10 — CHANGES_REQUESTED** [review]
@baothientran Two comments here. Let me know when ready

File: ThirdParty.json (hunk 2 above)
Comment (baothientran) [reply ThirdParty.json:102]: I updated the tidy-html in the new commit. I think we use only tidy-html5 and MikkTSpace in the plugin, plus all the third-parties come with Cesium Native

**Comment by shehzan10** [comment]
Thanks @baothientran!

**Pull request merged**